src/reduxCycles.js

```diff
@@ -0,0 +1,64 @@
+import xs from 'xstream';
+import {adapt} from '@cycle/run/lib/adapt';
+
+let actionListener = null;
+let stateListener = null;
```

Comment:
This is why I had used a store enhancer in my implementation. I like this approach though as it allows for injectable dependencies as well and is simpler than an enhancer that decorates the store contract.

Another plus to this approach is that it is rather simple to pass an adapter into createCycleMiddleware as an argument that allows for users to utilize other stream libraries.